fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/InOpValue.java

```diff
@@ -213,7 +213,6 @@ public InFn() {
         private static Value encapsulateInternal(@Nonnull final List<? extends Typed> arguments) {
             final Typed arg0 = arguments.get(0);
             final Type res0 = arg0.getResultType();
-            SemanticException.check(res0.isPrimitive(), SemanticException.ErrorCode.COMPARAND_TO_COMPARISON_IS_OF_COMPLEX_TYPE);
```

Comment:
The following is not dealt with correctly: Let's say you have a table with longs `t.x, t.y, t,z)`. The encapsulation for this expression should promote the parts individually, i.e. the `(x.t, x.y, x.z) IN (1, 2, 3)` needs to be encapsulated into
`(x.t, x.y, x.z) IN (PROMOTE 1 TO LONG, PROMOTE 2 TO LONG, PROMOTE 2 TO LONG)`. Similarly, the opposite needs to happen in the inverse case. The code here was written for primitive types and I am almost sure it cannot handle the complex case.